mujoco_warp/_src/smooth_test.py

```diff
@@ -361,17 +361,34 @@ def test_factor_solve_i(self, sparse):
     qM = np.zeros((mjm.nv, mjm.nv))
     mujoco.mj_fullM(mjm, qM, mjd.qM)
 
-    d.qLD.zero_()
+    d.qLD.fill_(wp.inf)
     if sparse:
-      d.qLDiagInv.zero_()
+      d.qLDiagInv.fill_(wp.inf)
 
     res = wp.zeros((1, mjm.nv), dtype=float)
     vec = wp.ones((1, mjm.nv), dtype=float)
 
     mjwarp._src.smooth.factor_solve_i(m, d, d.qM, d.qLD, d.qLDiagInv, res, vec)
-
     _assert_eq(res.numpy()[0], np.linalg.solve(qM, vec.numpy()[0]), "qM \\ 1")
 
+    if sparse:
```

Comment:
Nice - this is a great fix but I'm a bit confused by this test!

It seems like we are (correctly) updating our expectation that `factor_solve_i` (both dense and sparse) now should write to `L` - why are you explicitly running `factor_i_sparse` and `factor_i_dense` - shouldn't you just check `qLD` (and `qLDiagInv` if sparse) after runnning `factor_solve_i` above?

Reply:
updated the test:
- in the sparse case `qLD` and `qLDiagInv` are compared with the fields in `mjd`
- in the dense case `qLD` is compared to the result from `np.linalg.cholesky(qM)`